### PR TITLE
[FIX] account : tax lock date message not recomputed on state change

### PR DESCRIPTION
**Pull request by nmarotte**
Issue: When viewing an invoice with an `invoice_date` lower than the
`tax_lock_date`, a message appears on the top of the form to inform
 the user. However when posting and then resetting to draft, the `date`
 field was reset to `tax_lock_date + 1` but the `invoice_date` field
 wasn't, so it is saying a different date than the actual date

Steps to reproduce :
 1) Go to a draft invoice, set the the tax lock date and set the
 invoice date to a few days before the tax lock date
 2) The banner appears, post the entry
 3) The entry date is now `tax_lock_date + 1` but `invoice_date` didn't
 change
 4) Reset to draft, invoice_date still unchanged but no banner

Why is that a bug:
 The invoice date doesn't accurately represent the actual date when
 changed due to a tax lock date

opw-2585663

### CODE REVIEW

**Comment by jco-odoo** [comment]
@nmarotte I don't think I can agree with this, because it is the accounting date that counts.  (e.g. the invoice date on a vendor bill is the one from your vendor)  But @qdp-odoo might know better.

**Comment by nmarotte** [comment]
> @nmarotte I don't think I can agree with this, because it is the accounting date that counts. (e.g. the invoice date on a vendor bill is the one from your vendor) But @qdp-odoo might know better.

Oh sorry I didn't see this comment. Actually after testing more the issue is still there even with that fix so I'll have to look somewhere, I thought I opened the PR in draft sorry about that :sweat_smile: 

I'll drop another comment once it is ready

**Comment by nmarotte** [comment]
Okay I found a better approach, let's see if the tests are okay with it
Do you think it should be fine resetting the date to invoice_date when we change the state to draft ? I think this new commit would work better than the previous one, with less impact (only when resetting to draft) but I'm not sure if it's going to work accounting wise 😅

**Comment by nmarotte** [comment]
Hello @jco-odoo  (I don't know if I should ping the one you mentioned again since I've drastically changed the fix),

What do you think of the new changes ? I've tried minimizing the impact of the fix :smile:

File: addons/account/tests/test_account_move_entry.py [hunk 1]
```diff
@@ -485,3 +487,43 @@ def test_misc_prevent_unlink_posted_items(self):
...
+                          "Invoice date did not move")
+        self.assertFalse(self.test_move.tax_lock_date_message, msg="The banner disappeared, date was moved after tax_lock_date")
+
+        self.test_move.button_draft()
+
+        self.assertEqual(self.test_move.date, self.test_move.invoice_date,
+                          "After a Reset to Draft, just like before posting it, both field are synchronized due to `account.move._onchange_invoice_date`")
+        self.assertEqual(self.test_move.invoice_date, before_posting_invoice_date,
+                          "Invoice date still sleeping")
+        self.assertTrue(self.test_move.tax_lock_date_message, msg="Look who's back ! The banner telling you that the date will be moved")
```
Comment (smetl) [inline addons/account/tests/test_account_move_entry.py:529]: plz use `assertRecordValues` to compare record values

**Review by william-andre — CHANGES_REQUESTED** [review]
The banner is gone because there is not need to show it anymore.

The tax lock date only affects the accounting date, not the invoicing date.

We should close this PR IMO.

**Comment by tsb-odoo** [comment]
> The banner is gone because there is not need to show it anymore.
> 
> The tax lock date only affects the accounting date, not the invoicing date.
> 
> We should close this PR IMO.

Agreed. Note logged on chatter of opw-2585663

**Comment by nmarotte** [comment]
Closing PR : not a bug, but a feature

**Pull request closed without merging**